x-pack/filebeat/input/streaming/websocket.go

```diff
@@ -351,15 +351,13 @@ func connectWebSocket(ctx context.Context, cfg config, url string, log *logp.Log
 
 // calculateWaitTime calculates the wait time for the next attempt based on the exponential backoff algorithm.
 func calculateWaitTime(waitMin, waitMax time.Duration, attempt int) time.Duration {
-	// calculate exponential backoff
 	base := float64(waitMin)
-	backoff := base * math.Pow(2, float64(attempt-1))
+	exponential := base * math.Pow(1.5, float64(attempt-1))
 
-	// calculate jitter proportional to the backoff
-	maxJitter := float64(waitMax-waitMin) * math.Pow(2, float64(attempt-1))
-	jitter := rand.Float64() * maxJitter
+	// apply jitter as a fraction, up to 50% of the exponential
+	jitter := rand.Float64() * exponential * 0.5
+	waitTime := time.Duration(exponential + jitter)
```

Comment:
I'm not convinced of this. We've gone from jitter being proportionally related to the domain of the wait to being proportional to the calculated wait (which is going to be greater than the domain). ISTM that it should be proportional to the distance from the min and max, `jitter := rand.Float64() * (waitMax-exponential) * (exponential-waitMin)` with exponential (preferably renamed to something like `base` or reverted to `backoff` to be more informative, being defined as `max(waitMin, min(waitMax, base * math.Pow(1.5, float64(attempt-1))))`). Note also that there are sigmoid functions for this that obviate the need for the min/max clamping.

However, see overall comment.